titiler/errors.py

```diff
@@ -33,8 +43,10 @@ class BadRequestError(TilerError):
     TileOutsideBounds: status.HTTP_404_NOT_FOUND,
     TileNotFoundError: status.HTTP_404_NOT_FOUND,
     MosaicNotFoundError: status.HTTP_404_NOT_FOUND,
+    NoAssetFoundError: status.HTTP_404_NOT_FOUND,
     RasterioIOError: status.HTTP_404_NOT_FOUND,
-    InvalidBandName: status.HTTP_404_NOT_FOUND,
+    MissingAssets: status.HTTP_404_NOT_FOUND,
+    InvalidAssetName: status.HTTP_404_NOT_FOUND,
```

Comment:
same ?

Reply:
This could either be 400 or 404.  You can argue that it should be 404 because the asset doesn't exist in the STAC item, you can also argue that the caller should know what assets are present in the item so it should raise a 400.

Personally I'm leaning towards 400 but I don't think either is wrong.

Reply:
thanks! I'll keep 404 👌